tests/unit/lifecycle/LifecycleTask.spec.js

```diff
@@ -82,9 +82,16 @@ describe('lifecycle task helper methods', () => {
 
             const res1 = lct._filterRules(mBucketRules, item1, objTags);
             assert.strictEqual(res1.length, 3);
-            const expRes1 = getRuleIDs(mBucketRules.filter(rule =>
-                (rule.Filter.Prefix && rule.Filter.Prefix.startsWith('atask'))
-                || !rule.Filter.Prefix));
+            const expRes1 = getRuleIDs(mBucketRules.filter(rule => {
+                if (rule.Filter.Prefix &&
+                item1.Key.startsWith(rule.Filter.Prefix)) {
+                    return true;
+                }
+                if (!rule.Filter.Prefix) {
```

Comment:
Don't you want this rule check above the previous one? if checking for empty string it should work though, but if undefined it can cause issues.